src/Payments/Integrations/Stripe.php

```diff
@@ -189,11 +189,14 @@ private function get_button_locale() {
 	 * @param PaymentResult  $result  Result object for the payment.
 	 */
 	public function add_payment_request_order_meta( PaymentContext $context, PaymentResult $result ) {
-		// phpcs:ignore WordPress.Security.NonceVerification
-		$post_data = $_POST;
-		$_POST     = $context->payment_data;
-
-		WC_Stripe_Payment_Request::add_order_meta( $context->order->id, $context->payment_data );
-		$_POST = $post_data;
+		$data = $context->payment_data;
+		if ( $data['payment_request_type'] && 'stripe' === $context->payment_method ) {
```

Comment:
is `payment_request_type` used anymore?

Reply:
Yes, it's passed through via the `$_POST` switcheroo to the stripe extension. I just check if it's present here first to determine whether the switcheroo is needed.

Reply:
In that case, check if `! empty` or it will show a notice for other gateways.

Reply:
```suggestion
		if ( ! empty( $data['payment_request_type'] ) && 'stripe' === $context->payment_method ) {
```

Reply:
Oh ya, missed that, will add.